packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -184,9 +184,27 @@ class CoverageCollector extends TestWatcher {
   }
 }
 
-Future<Map<String, dynamic>> collect(Uri serviceUri, bool Function(String) libraryPredicate) async {
+Future<Map<String, dynamic>> collect(Uri serviceUri, bool Function(String) libraryPredicate, { bool waitPaused = false }) async {
+  const int _kPausePollLimit = 5;
   final VMService vmService = await VMService.connect(serviceUri, compression: CompressionOptions.compressionOff);
   await vmService.getVM();
+  if (waitPaused) {
+    int i = 0;
+    Duration waitDuration = const Duration(seconds: 0);
+    while (i < _kPausePollLimit) {
+      await Future<void>.delayed(waitDuration);
+      for (Isolate isolateRef in vmService.vm.isolates) {
```

Comment:
So this waits for *any* isolate to be paused?

Reply:
Update to only wait for the correct test isolate